pkg/systemd/services/services.jsx

```diff
@@ -481,14 +484,20 @@
                                             if (hasExtraEntries)
                                                 newState.unit_by_path = unit_by_path;
 
-                                            newState.loadingUnits = false;
                                             newState.isFullyLoaded = true;
+                                            this.props.setIsLoading(false);
 
                                             this.setState(newState);
                                             this.processFailedUnits();
                                         });
-                            }, ex => this.setState({ error: cockpit.format(_("Listing unit files failed: $0"), ex.toString()), loadingUnits: false }));
-                }, ex => this.setState({ error: cockpit.format(_("Listing units failed: $0"), ex.toString()), loadingUnits: false }));
+                            }, ex => {
+                                this.props.isLoading(false);
+                                this.setState({ error: cockpit.format(_("Listing unit files failed: $0"), ex.toString()) });
+                            });
+                }, ex => {
+                    this.props.isLoading(false);
+                    this.setState({ error: cockpit.format(_("Listing units failed: $0"), ex.toString()) });
+                });
```

Comment:
These 8 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18005-20221207-164851-4022857c-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)